models.py

```diff
@@ -37,42 +65,16 @@ def generate_compiled_segmentation_model(model_name, model_parameters, num_class
     else:  # to guarantee the V1 config files still work
         model_parameters['input_shape'] = (None, None, 1)
 
-    # These are the only optimizer currently supported
-    if optimizer.lower() == 'adam':
-        opt_fn = Adam()
-    elif optimizer.lower() == 'sdg':
-        opt_fn = SGD()
-    elif optimizer.lower() == 'adagrad':
-        opt_fn = Adagrad()
-    elif optimizer.lower() == 'adamax':
-        opt_fn = Adamax()
-    elif optimizer.lower() == 'ftrl':
-        opt_fn = Ftrl()
-    elif optimizer.lower() == 'nadam':
-        opt_fn = Nadam()
-    elif optimizer.lower() == 'rmsprop':
-        opt_fn = RMSprop()
-    else:
-        raise NameError("Optimizer not supported")
+    # Select the optimizer as a function of the name in the config file
+    optimizer_fn = optimzer_dict[optimizer.lower()]
+
+    # Select the loss function  as a function of the name in the config file
+    loss_fn = loss_dict[loss.lower()]
```

Comment:
add if-else to make sure `[loss.lower()]` is in `loss_dict`, proceed to assignment if true but throw helpful error message if choosing a currently unsupported option